src/test/java/com/aws/greengrass/security/SecurityServiceTest.java

```diff
@@ -29,80 +37,128 @@
-        when(mockKeyProvider.supportedKeyType()).thenReturn("FILE");
+    void GIVEN_key_service_provider_registered_WHEN_deregister_a_different_instance_THEN_the_provider_not_removed()
+            throws Exception {
+        when(mockKeyProvider.supportedKeyType()).thenReturn("PKCS11");
         service.registerCryptoKeyProvider(mockKeyProvider);
         CryptoKeySpi providerB = mock(CryptoKeySpi.class);
-        when(providerB.supportedKeyType()).thenReturn("file");
+        when(providerB.supportedKeyType()).thenReturn("pkcs11");
         service.deregisterCryptoKeyProvider(providerB);
-        assertThat(service.getCryptoKeyProviderMap(), IsMapWithSize.aMapWithSize(1));
-        assertThat(service.getCryptoKeyProviderMap(), IsMapContaining.hasEntry(new CaseInsensitiveString("file"),
-                mockKeyProvider));
+        assertThat(service.getCryptoKeyProviderMap(), IsMapWithSize.aMapWithSize(2));
+        assertThat(service.getCryptoKeyProviderMap(),
+                IsMapContaining.hasEntry(new CaseInsensitiveString("pkcs11"), mockKeyProvider));
     }
 
     @Test
-    void GIVEN_key_service_provider_registered_WHEN_get_key_managers_THEN_delegate_call_to_service_provider() throws Exception {
-        when(mockKeyProvider.supportedKeyType()).thenReturn("FILE");
-        String keyUri = "file:///path/to/file";
+    void GIVEN_key_service_provider_registered_WHEN_get_key_managers_THEN_delegate_call_to_service_provider()
+            throws Exception {
+        when(mockKeyProvider.supportedKeyType()).thenReturn("PKCS11");
+        String keyUri = "pkcs11:object=key-label";
+        String certificateUri = "file:///path/to/certificate";
         KeyManager[] mockKeyManagers = {mock(KeyManager.class)};
-        when(mockKeyProvider.getKeyManagers(keyUri)).thenReturn(mockKeyManagers);
+        when(mockKeyProvider.getKeyManagers(keyUri, certificateUri)).thenReturn(mockKeyManagers);
         service.registerCryptoKeyProvider(mockKeyProvider);
-        KeyManager[] keyManagers = service.getKeyManagers(keyUri);
+        KeyManager[] keyManagers = service.getKeyManagers(keyUri, certificateUri);
         assertThat(keyManagers, Is.is(mockKeyManagers));
     }
 
     @Test
-    void GIVEN_key_service_provider_not_registered_WHEN_get_key_managers_THEN_throw_exception () {
-        assertThrows(ServiceUnavailableException.class, () -> service.getKeyManagers("file:///path/to/file"));
+    void GIVEN_key_service_provider_not_registered_WHEN_get_key_managers_THEN_throw_exception() {
+        assertThrows(ServiceUnavailableException.class,
+                () -> service.getKeyManagers("pkcs11:object=key-label", "file:///path/to/certificate"));
+    }
+
+    @Test
+    void GIVEN_key_and_cert_uri_WHEN_get_key_managers_from_default_THEN_succeed() throws Exception {
+        Path certPath =
+                EncryptionUtilsTest.generateCertificateFile(2048, true, resourcePath.resolve("certificate.pem"));
```

Comment:
Do you know if we support other RSA key lengths, other than 2048?

Reply:
yes, we do. 1024 or 512 are fine as well.